api/views.py

```diff
@@ -27,21 +27,47 @@ def favorites(self, request, pk=None):
     favorites = user.grants.all()
     serializer = GrantSerializer(favorites, many=True)
     return Response(serializer.data)
-  
-  # @action(detail=True, methods=['post'])
-  # def favorites(self, request, pk=None, pk_grant=None):
-  #   user = self.get_object()
-  #   favorite = Grant.objects.get(pk=pk_grant)
-  #   user.grants.add(favorite)
-  #   serializer = GrantSerializer(favorite, many=False)
-  #   return JsonResponse(serializer.data)
+
 
 class GrantViewSet(viewsets.ModelViewSet):
   queryset = Grant.objects.all()
   serializer_class = GrantSerializer
 
   def list(self, request):
-    serializer = GrantSerializer(self.queryset, many=True)
+    education = request.GET.get('education','')
+    gender = request.GET.get('gender','')
+    state = request.GET.get('state','')
+    lgbt = request.GET.get('lgbt','')
+    veteran = request.GET.get('veteran','')
+    immigrant = request.GET.get('immigrant','')
+    unfiltered_ethnicity = request.GET.get('ethnicity','')
+    
+    q = {}
+
+    if education != '':
+      q.update({'education': education})
+
+    if gender != '':
+      q.update({'gender': gender})
+
+    if state != '':
+      q.update({'state': state})
+
+    if lgbt != '':
+      q.update({'lgbt': lgbt})
+
+    if unfiltered_ethnicity != '':
+      ethnicity = sorted(unfiltered_ethnicity.replace('[','').replace(']','').split(','))
+      q.update({'ethnicity': ethnicity})
+    
+    if veteran != '':
+      q.update({'veteran': veteran})
+
+    if immigrant != '':
+      q.update({'immigrant': immigrant})
+    
```

Comment:
Potential Refactor: With the exception of unfilltered_ethnicity, all of these cases are handled the same way. Can you clean this up to have an if statement that 
1) checks if a query parameter is present (`query_param in request.GET`)
2) adds the key/value pair to the hash/dictionary

This could even be pulled into a separate helper method